src/app/Main/Body/QueryPage/ChangeDisplFeatures.service.tsx

```diff
@@ -1,11 +1,10 @@
-import { ValueType } from "react-select";
 import { FilterType } from "../../../Shared/Model/Filter.model";
 
 export function handleChangeDisplayedFeatures(
-    selectedOption: ValueType<{value: string, label: string}>
+    selectedOption: {value: string, label: string}
 ): FilterType {
-    if (selectedOption) {
-        const selectedFeature: string = (Object.values(selectedOption)[0]);
+    if (selectedOption !== undefined && selectedOption.value !== undefined) {

```

Comment:
reject: selectedOption is typed and it says itself and the value cannot be undefined, so you just changed the type but not the values that can stand behind the type: see comments below for solution

Reply:
This function has been changed to export function chooseSelectedDisplayedFeaturesService.  Can you please look into these functions again and say if changes are still needed?

Reply:
looks good, but service is kind of obsolete now and can be replaced by:
`selectedFeature = selectedOption  === null ? "" : selectedOption.value`
or completely removed if your selector component returns: `onSelectionChange(value: string)` with value = "" for no selection.